frontend/src/views/student/tournament/MyTournamentsView.vue

```diff
@@ -0,0 +1,348 @@
+      await this.$store.dispatch('error', error);
+    }
+    await this.$store.dispatch('clearLoading');
+  }
+
+  sortById(a: Tournament, b: Tournament) {
+    if (a.id && b.id) return a.id > b.id ? 1 : -1;
+    else return 0;
+  }
+
+  openTournamentDashboard(tournament: Tournament) {
+    if (tournament) return '/student/tournament?id=' + tournament.id;
+  }
+
+  newTournament() {
+    this.currentTournament = new Tournament();
+    this.createTournamentDialog = true;
+  }
+
+  async onCreateTournament(tournament: Tournament) {
+    this.tournaments.unshift(tournament);
+    this.createTournamentDialog = false;
+    this.currentTournament = null;
+  }
+
+  onCloseDialog() {
+    this.createTournamentDialog = false;
+    this.currentTournament = null;
+  }
+
+  editTournament(tournamentToEdit: Tournament) {
+    this.currentTournament = tournamentToEdit;
+    this.editTournamentDialog = true;
+  }
+
+  async onEditTournament(tournament: Tournament) {
+    this.currentTournament = tournament;
+    try {
+      this.tournaments = await RemoteServices.getUserTournaments();
+    } catch (error) {
+      await this.$store.dispatch('error', error);
+    }
+    this.editTournamentDialog = false;
+    this.currentTournament = null;
+  }
+
+  onCloseEditDialog() {
+    this.editTournamentDialog = false;
+    this.currentTournament = null;
+  }
+
+  getStateColor(tournament: Tournament) {
+    if (tournament.id && this.closedTournamentsId.includes(tournament.id))
+      return 'orange';
+    else if (!tournament.isCanceled) return 'green';
+    else return 'red';
+  }
+
+  getStateName(tournament: Tournament) {
+    if (tournament.id && this.closedTournamentsId.includes(tournament.id))
+      return 'FINISHED';
+    else if (!tournament.isCanceled) return 'AVAILABLE';
+    else return 'CANCELLED';
+  }
+
+  getEnrolledColor(enrolled: string) {
+    if (enrolled) return 'green';
+    else return 'red';
+  }
+
+  getEnrolledName(enrolled: string) {
+    if (enrolled) return 'YOU ARE IN';
+    else return 'YOU NEED TO JOIN';
+  }
+
+  getPrivateColor(privateTournament: boolean) {
+    if (privateTournament) return 'red';
+    else return 'green';
+  }
+
+  getPrivateName(privateTournament: boolean) {
+    if (privateTournament) return 'PRIVATE';
+    else return 'PUBLIC';
+  }
+
+  isNotCanceled(tournamentToCancel: Tournament) {
+    return !tournamentToCancel.isCanceled;
+  }
+
+  async cancelTournament(tournamentToCancel: Tournament) {
+    if (confirm('Are you sure you want to cancel this tournament?')) {
+      const participants = tournamentToCancel.participants;
+      tournamentToCancel.participants = [];
+      try {
+        await RemoteServices.cancelTournament(tournamentToCancel);
+      } catch (error) {
+        await this.$store.dispatch('error', error);
+        tournamentToCancel.participants = participants;
+        return;
+      }
+      tournamentToCancel.isCanceled = true;
+      tournamentToCancel.participants = participants;
```

Comment:
It is not elegant this unset and set of the participants

Reply:
É necessário, caso contrário origina problemas ao nivel da base de dados

Reply:
What type of problems?